errors/errors.go

```diff
@@ -1,22 +1,89 @@
+	})
+	// ErrMalformedJSON lint comment
+	ErrMalformedJSON = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "MALFORMED_JSON",
+		Message:        "JSON sent by a client could not be parsed by the server",
+		Description:    "This error indicates a client has sent malformed JSON.",
+		HTTPStatusCode: http.StatusBadRequest,
+	})
+	// ErrUpdating lint comment
+	ErrUpdating = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "UPDATING",
+		Message:        "An error has occurred while updating the TUF repository: %s",
+		Description:    "If an error occurs while attempting to update the storage with new TUF data, this error is returned.",
+		HTTPStatusCode: http.StatusInternalServerError,
+	})
+	// ErrMetadataNotFound lint comment
+	ErrMetadataNotFound = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "METADATA_NOT_FOUND",
+		Message:        "You have requested metadata that does not exist.",
+		Description:    "This error is returned when the user requests metadata that is not known to the server.",
+		HTTPStatusCode: http.StatusNotFound,
+	})
+	// ErrMalformedUpload lint comment
+	ErrMalformedUpload = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "MALFORMED_UPLOAD",
+		Message:        "The body of your request is malformed.",
+		Description:    "This error is raised when a user uploads new TUF data and we are unable to parse it as multipart/form-data.",
+		HTTPStatusCode: http.StatusBadRequest,
+	})
+	// ErrGenericNotFound lint comment
+	ErrGenericNotFound = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "GENERIC_NOT_FOUND",
+		Message:        "You have requested a resource that does not exist.",
+		Description:    "This error is returned when the user requests a non-specific resource that is not known to the server.",
+		HTTPStatusCode: http.StatusNotFound,
+	})
+	// ErrNoCryptoService lint comment
+	ErrNoCryptoService = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "NO_CRYPTOSERVICE",
+		Message:        "The server does not have a signing service configured.",
+		Description:    "This error is returned if no signing service has been configured for the server and it has been asked to perform an operation that requires either signing, or key generation.",
+		HTTPStatusCode: http.StatusInternalServerError,
+	})
+	// ErrUnauthorized lint comment
+	ErrUnauthorized = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "UNAUTHORIZED",
+		Message:        "The server does not have a signing service configured.",
+		Description:    "This error is returned if no signing service has been configured for the server and it has been asked to perform an operation that requires either signing, or key generation.",
+		HTTPStatusCode: http.StatusUnauthorized,
+	})
+	// ErrUnknown lint comment
+	ErrUnknown = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "GENERAL_UNKNOWN",
```

Comment:
Detail gets added to the json error response. Typically, this can be some other structure, describing the failure. It isn't meant to augment the error message.